src/main/java/ch/njol/skript/patterns/SkriptPattern.java

```diff
@@ -86,4 +86,65 @@ public static String[] getKeywords(PatternElement first) {
 		return keywords.toArray(new String[0]);
 	}
 
+	/**
+	 * @return the size of the {@link MatchResult#expressions} array
+	 * from a match.
+	 */
+	public int countTypes() {
+		return expressionAmount;
+	}
+
+	/**
+	 * Count the maximum amount of non-null types in this pattern,
+	 * i.e. the maximum amount of non-null values in the {@link MatchResult#expressions}
+	 * array from a match.
+	 *
+	 * @see #countTypes() for the amount of nullable values
+	 * in the expressions array from a match.
+	 */
+	public int countNonNullTypes() {
+		return countNonNullTypes(first);
+	}
+
+	/**
+	 * Count the maximum amount of non-null types in the given pattern,
+	 * i.e. the maximum amount of non-null values in the {@link MatchResult#expressions}
+	 * array from a match.
+	 */
+	private static int countNonNullTypes(PatternElement patternElement) {
+		int count = 0;
+
+		// Iterate over all consequent pattern elements
+		while (patternElement != null) {
+			if (patternElement instanceof ChoicePatternElement) {
+				// Keep track of the max type count of each component
+				int max = 0;
+
+				for (PatternElement component : ((ChoicePatternElement) patternElement).getPatternElements()) {
+					int componentCount = countNonNullTypes(component);
+					if (componentCount > max) {
+						max = componentCount;
+					}
```

Comment:
```suggestion
					if (componentCount > max)
						max = componentCount;
```

Reply:
I personally don't like omitting them (because of readability), and our code style guide doesn't require omitting them either (it merely states they may be omitted)